internal/cimfs/cim_writer_windows.go

```diff
@@ -0,0 +1,261 @@
+package cimfs
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"unsafe"
+
+	"github.com/Microsoft/go-winio"
+	"github.com/Microsoft/hcsshim/internal/winapi"
+	"github.com/pkg/errors"
+	"golang.org/x/sys/windows"
+)
+
+// CimFsWriter represents a writer to a single CimFS filesystem instance. On disk, the
+// image is composed of a filesystem file and several object ID and region files.
+type CimFsWriter struct {
+	// name of this cim. Usually a <name>.cim file will be created to represent this cim.
+	name string
+	// handle is the CIMFS_IMAGE_HANDLE that must be passed when calling CIMFS APIs.
+	handle winapi.FsHandle
+	// name of the active file i.e the file to which we are currently writing.
+	activeName string
+	// stream to currently active file.
+	activeStream winapi.StreamHandle
+	// amount of bytes that can be written to the activeStream.
+	activeLeft int64
+}
+
+// creates a new cim image. The handle returned in the `cim.handle` variable can then be
```

Comment:
This is kind of a weird comment as I assume we'll be calling this externally from this package, but then we won't actually be able to directly access the handle as it's not exported.

Reply:
Updated the comment.